### PR TITLE
[Bug](date function) `from_unixtime` return wrong result

### LINKED ISSUE #11409: [Bug] `from_unixtime` return wrong result
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

`from_unixtime` will be fold in FE by rule `RewriteFromUnixTimeRule`. But this rewrite may produce an incorrect value. 

### What You Expected?

Fix

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #11409

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: regression-test/suites/query/sql_functions/datetime_functions/test_date_function.groovy [hunk 1]
```diff
@@ -253,4 +253,22 @@ suite("test_date_function", "query") {
...
+                `id` INT NOT NULL COMMENT "用户id",
+                `update_time` INT NOT NULL COMMENT "数据灌入日期时间"
+            ) ENGINE=OLAP
+            UNIQUE KEY(`id`)
+            DISTRIBUTED BY HASH(`id`)
+            PROPERTIES("replication_num" = "1");
+        """
+    sql """ insert into ${tableName} values (1, 1659344431) """
+
+    qt_sql """ SELECT id,FROM_UNIXTIME(update_time,"%Y-%m-%d") FROM ${tableName} WHERE FROM_UNIXTIME(update_time,"%Y-%m-%d") = '2022-08-01'; """
```
Comment (yiguolei) [inline regression-test/suites/query/sql_functions/datetime_functions/test_date_function.groovy:271]: add more test case, >, >= , = , < , <= , like

**Comment by morrySnow** [comment]
the best way to do that is change left hand value according to binary predicate's operator and date format in left hand. So please add a TODO to explain it and fix it later.

File: regression-test/suites/query/sql_functions/datetime_functions/test_date_function.groovy (hunk 1 above)
Comment (Gabriel39) [reply regression-test/suites/query/sql_functions/datetime_functions/test_date_function.groovy]: done

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**